doc/code/templates.rst

```diff
@@ -0,0 +1,115 @@
+.. _templates:
+
+Templates
+=========
+
+**Module name:** :mod:`pennylane.templates`
+
+.. currentmodule:: pennylane.templates
+
+This module provides a growing library of templates of common quantum
+machine learning circuit architectures that can be used to easily build,
+evaluate, and train more complex quantum machine learning models. In the
+quantum machine learning literature, such architectures are commonly known as an
+**ansatz**.
+
+PennyLane conceptually distinguishes two types of templates, layer architectures and input embeddings:
+
+* Layer architectures in :mod:`pennylane.templates.layers` define blocks of gates that are repeated like the
+  layers in a neural network. They usually contain only trainable parameters.
+
+* Embeddings in :mod:`pennylane.templates.embeddings` encode input features into the quantum state of the
+  circuit. These embeddings can also be learnable, or depend on trainable parameters.
+
+Each template has a utilities function in :mod:`pennylane.templates.parameters` which generates a randomly
+sampled array for the trainable parameters fed into the template.
```

Comment:
Are these utility functions simply for providing initial numeric values for the parameters? Or can it also be used to generate pytorch/tf variables for training?

Reply:
For now they are just returning numpy arrays. They could be turned into tf/pytorch ones by casting every entry of the returned list into a tensor. 

An idea for future features?

Reply:
:+1: